aptos-move/framework/aptos-stdlib/sources/cryptography/groups.move

```diff
@@ -0,0 +1,749 @@
+        let point_7g_from_comp = std::option::extract(&mut deserialize_element_compressed<BLS12_381_G1>(x"b7fc7e62705aef542dbcc5d4bce62a7bf22eef1691bef30dac121fb200ca7dc9a4403b90da4501cfee1935b9bef32899"));
+        assert!(element_eq(&point_7g_from_comp, &point_7g_from_uncomp), 1);
+
+        // Point multiplication by scalar.
+        let point_7g_calc = element_scalar_mul(&scalar_7, &point_g);
+        assert!(element_eq(&point_7g_calc, &point_7g_from_comp), 1);
+        assert!(x"b7fc7e62705aef542dbcc5d4bce62a7bf22eef1691bef30dac121fb200ca7dc9a4403b90da4501cfee1935b9bef328191c1a98287eec115a8cb0a1cf4968c6fd101ca4593938d73918dd8e81471d8a3ac4b38930aed539564436b6a4baad8d10" == serialize_element_uncompressed(&point_7g_calc), 1);
+        assert!(x"b7fc7e62705aef542dbcc5d4bce62a7bf22eef1691bef30dac121fb200ca7dc9a4403b90da4501cfee1935b9bef32899" == serialize_element_compressed(&point_7g_calc), 1);
+
+        // Point double.
+        let point_2g = element_scalar_mul(&scalar_2, &point_g);
+        let point_double_g = element_double(&point_g);
+        assert!(element_eq(&point_2g, &point_double_g), 1);
+
+        // Point negation.
+        let point_minus_7g_calc = element_neg(&point_7g_calc);
+        assert!(x"b7fc7e62705aef542dbcc5d4bce62a7bf22eef1691bef30dac121fb200ca7dc9a4403b90da4501cfee1935b9bef32819" == serialize_element_compressed(&point_minus_7g_calc), 1);
+        assert!(x"b7fc7e62705aef542dbcc5d4bce62a7bf22eef1691bef30dac121fb200ca7dc9a4403b90da4501cfee1935b9bef328198f9067d78113ed5f734fb2e1b497e52013da0c9d679a592da735f6713d2eed2913f9c11208d2e1f455b0c9942f647309" == serialize_element_uncompressed(&point_minus_7g_calc), 1);
+
+        // Point addition.
+        let point_9g = element_scalar_mul(&scalar_9, &point_g);
+        let point_2g = element_scalar_mul(&scalar_2, &point_g);
+        let point_2g_calc = element_add(&point_minus_7g_calc, &point_9g);
+        assert!(element_eq(&point_2g, &point_2g_calc), 1);
+
+        // Multi-scalar multiplication.
+        let point_14g = element_scalar_mul(&scalar_from_u64<BLS12_381_G1>(14), &point_g);
+        let scalar_1 = scalar_from_u64<BLS12_381_G1>(1);
+        let scalar_2 = scalar_from_u64<BLS12_381_G1>(2);
+        let scalar_3 = scalar_from_u64<BLS12_381_G1>(3);
+        let point_2g = element_scalar_mul(&scalar_2, &point_g);
+        let point_3g = element_scalar_mul(&scalar_3, &point_g);
+        let scalars = vector[scalar_1, scalar_2, scalar_3];
+        let points = vector[point_g, point_2g, point_3g];
+        let point_14g_calc = element_multi_scalar_mul(&scalars, &points);
+        assert!(element_eq(&point_14g, &point_14g_calc), 1);
+
+        // Hash to group.
+        let point = hash_to_element<BLS12_381_G1>(x"1234");
+        assert!(element_eq(&group_identity<BLS12_381_G1>(), &element_scalar_mul(&scalar_from_u64<BLS12_381_G1>(0), &point)), 1);
```

Comment:
What does this `assert` achieve?

Reply:
Let's move the hash-to-group feature into a separate PR!